core/src/test/scala/com/nvidia/spark/rapids/tool/tuning/ProfilingAutoTunerSuite.scala

```diff
@@ -1883,6 +1874,185 @@ We recommend using nodes/workers with more memory. Need at least 17496MB memory.
     compareOutput(expectedResults, autoTunerOutput)
   }
 
+  // Map of `Spark Version` -> `memoryOverheadLabel`
+  private val MISSING_MEMORY_OVERHEAD_k8s_TEST_CASES = Seq(
+    "3.5.0" -> "spark.executor.memoryOverheadFactor",
+    "3.2.1" -> "spark.kubernetes.memoryOverheadFactor"
+  )
+
+  // This UT sets a custom spark-property "spark.master" pointing to a spark
+  // k8s value. The Autotuner should detect that the spark-master is k8s and
+  // comments on the missing memoryOverhead value since pinned pool is set.
+  MISSING_MEMORY_OVERHEAD_k8s_TEST_CASES.foreach { case (sparkVersion, memoryOverheadLabel) =>
+    test(s"missing memoryOverhead comment is included for k8s with pinned pool " +
+      s"[sparkVersion=$sparkVersion]") {
+      val logEventsProps: mutable.Map[String, String] =
+        mutable.LinkedHashMap[String, String](
+          "spark.master" -> "k8s://https://my-cluster-endpoint.example.com:6443",
+          "spark.executor.cores" -> "16",
+          "spark.executor.instances" -> "1",
+          "spark.executor.memory" -> "80g",
+          "spark.executor.resource.gpu.amount" -> "1",
+          "spark.executor.instances" -> "1",
+          "spark.sql.shuffle.partitions" -> "200",
+          "spark.sql.files.maxPartitionBytes" -> "1g",
+          "spark.task.resource.gpu.amount" -> "0.0625",
+          "spark.rapids.memory.pinnedPool.size" -> "5g",
+          "spark.rapids.sql.enabled" -> "true",
+          "spark.plugins" -> "com.nvidia.spark.SQLPlugin",
+          "spark.rapids.sql.concurrentGpuTasks" -> "4")
+      val dataprocWorkerInfo = buildGpuWorkerInfoAsString()
+      val infoProvider = getMockInfoProvider(8126464.0, Seq(0), Seq(0.004), logEventsProps,
+        Some(sparkVersion))
+      val clusterPropsOpt = ProfilingAutoTunerConfigsProvider
+        .loadClusterPropertiesFromContent(dataprocWorkerInfo)
+      val platform = PlatformFactory.createInstance(PlatformNames.ONPREM, clusterPropsOpt)
+      val autoTuner: AutoTuner = ProfilingAutoTunerConfigsProvider
+        .buildAutoTunerFromProps(dataprocWorkerInfo, infoProvider,
+          platform)
+      val (_, comments) = autoTuner.getRecommendedProperties()
+      val expectedComment =
+        s"'$memoryOverheadLabel' must be set if using 'spark.rapids.memory.pinnedPool.size'."
+      assert(comments.exists(_.comment == expectedComment))
+    }
+  }
+
+  // This UT sets a custom spark-property "spark.master" pointing to a spark
+  // k8s value. The Autotuner should detect that the spark-master is k8s and
+  // should not comment on the missing memoryOverhead value since pinned pool is not set.
+  test(s"missing memoryOverhead comment is not included for k8s without pinned pool") {
```

Comment:
can we include spark-version in the test comment to differentiate between the 2 test runs?

Reply:
Included spark version in the test name since spark version is dynamic and set during runtime.